### PR TITLE
Feature/update user details

### PR DESCRIPTION
**Pull request by swetshaw**
### [DE-6](https://trello.com/c/h9huNS87)

- Allows user to update their personal info after Login.

- User can add and delete addresses from their account.

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -5,27 +5,31 @@ const schema = mongoose.Schema;
...
+      pin: { type: Number, default: 0 },
     },
   ],
 
   gender: {
     type: String,
+    default: " ",
   },
 
   contactno: {
```
Comment (debnathSD) [inline models/User.js:30]: If we are having contact number as a required field, then why do we need a `default` value for that?

File: routes/api/product.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * @route   GET /api/v1/auth/getProductsDetails
+ * @desc    Get Products Details
+ * @access  Public
+ */
+
+router.get("/getProductsDetails", (req, res) => {
+    
+  
+  // Find a User By Email
+  Product.find({ }).then((product) => {
```
Comment (debnathSD) [inline routes/api/product.js:18]: How is it searching by `mail od` , couldn't get that part

File: server.js [hunk 3]
```diff
@@ -33,6 +34,7 @@ app.get("/", (req, res) => res.send(`Hey Base Route!!!`));
 // Use Routes
 app.use("/api/v1/auth", auth);
 app.use("/api/v1/auth", user);
+app.use("/api/v1/auth", address);
```
Comment (debnathSD) [inline server.js:37]: Can we have products in a separate `/api/v1/product` route please? @swetshaw

File: models/User.js (hunk 1 above)
Comment (swetshaw) [reply models/User.js:30]: Yeah.. Will change it.

File: server.js (hunk 3 above)
Comment (swetshaw) [reply server.js:37]: Yeah Sure.

File: routes/api/product.js (hunk 2 above)
Comment (debnathSD) [reply routes/api/product.js:18]: Ignoring product route as it is out of scope

**Pull request merged**